### ISSUE TITLE
Strange behavior on mobile chrome and modals

### ISSUE THREAD

**Issue by ahoiroman** [tailwindlabs/headlessui#1303]
**What package within Headless UI are you using?**

@headlessui/vue

**What version of that package are you using?**

1.5.0

**What browser are you using?**

Chrome on mobile and Chrome on Android Emulator

**Reproduction URL**

https://ruytgy.csb.app/

https://user-images.githubusercontent.com/23051150/161567687-36c269a3-6927-4c51-8c57-3aac55059d83.mp4

**Describe your issue**

Hello everybody,

I recently got informed that mobile users on chrome/android are experiencing a little "funny" behavior on modals. The autofocus does not work as expected and the view is always scrolled to the bottom.

This happens on modals, which are embedded at the end of the page.

### Code example

This is the code of `App.vue` in this example:

```vue
<template>
  <div class="h-screen">
    <div class="h-96 bg-red-500 flex justify-center items-center">
      <span class="text-white">Scroll down...</span>
    </div>
    <div class="h-96 bg-red-500 flex justify-center items-center">
      <span class="text-white">Just a little bit more...</span>
    </div>
    <div class="h-96 bg-red-500 flex justify-center items-center">
      <span class="text-white">You are almost there...</span>
    </div>
    <div class="h-96 bg-red-500 flex justify-center items-center">
      <span class="text-white"
        >Here we are. Now scroll down and click the button...</span
      >
    </div>

    <SimpleModal ref="mymodal">
      <template #dialog>
        <InputField name="test1" v-model="value1" />
        <InputField name="test2" v-model="value2" />
        <InputField name="test3" v-model="value3" />
        <InputField name="test4" v-model="value4" />
        <InputField name="test5" v-model="value5" />
        <InputField name="test6" v-model="value6" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
        <InputField name="test7" v-model="value7" />
      </template>
    </SimpleModal>
    <div class="flex justify-center items-center mt-4">
      <button class="border p-4" @click="$refs.mymodal.openModal">
        Open Modal
      </button>
    </div>
  </div>
</template>
<script>
import { ref } from "vue";
import InputField from "./components/InputField.vue";
import SimpleModal from "./components/SimpleModal.vue";

export default {
  components: {
    SimpleModal,
    InputField,
  },

  setup() {
    const isOpen = ref(false);

    return {
      isOpen,
      closeModal() {
        isOpen.value = false;
      },
      openModal() {
        isOpen.value = true;
      },
    };
  },
  data() {
    return {
      value1: null,
      value2: null,
      value3: null,
      value4: null,
    };
  },
};
</script>
```

### Ugly workaround
For now I created a little ugly workaround: Before the modal is being opened, use

```js
window.scroll(0,0);
```

to scroll to the top. That's not really what I want, because if the person closes the modal, the person has to scroll down all the way again.

**Comment by bobinska-dev** [comment]
I have the same issue!!! 
NextJS latest (12.1.5)
Version HeadlessUI React 1.6.0
It worked until a couple of days ago

**Comment by ahoiroman** [comment]
I "solved" this bug by adding another workaround:

I created an overflow-scroll to the modal's content container. This is not a very nice solution, but it works for now.

```html
<template>
    <DialogTitle  ...>
        ...
    </DialogTitle>
    <div class="modal-container overflow-y-auto">
        ...
    </div>
</template>
```

```css
    .modal-container {
        --dialog-margin-y: 6rem;
        --dialog-max-height: 100vh;
        max-height: calc(var(--dialog-max-height) - calc(2 * var(--dialog-margin-y)));
        overflow-y: auto;
    }
 ```

**Comment by bobinska-dev** [comment]
Okay I went back 2 Versions just now and that solved It for me as well...

**Comment by ahoiroman** [comment]
Could you please tell me, which exact version you are using now? I moved back and it did not solve the issue for me. Thanks in advance :-)

**Comment by bobinska-dev** [comment]
"@headlessui/react": "1.4.0"

**Comment by ahoiroman** [comment]
Thanks. Maybe something @RobinMalfait should know about? Seems like something that could easily be missed by tests and QA?

**Comment by ahoiroman** [comment]
Tested it on 1.6.1 without success :-(

**Comment by ahoiroman** [comment]
Sandbox for 1.6.1: https://codesandbox.io/s/ruytgy?file=/package.json

**Comment by thecrypticace** [comment]
This appears to be a weird bug involving `overflow: hidden` on the root, zero body margins, and a scrollable fixed container. I've created a reproduction that's entirely outside of Headless UI or Vue/React. It's a static HTML page with vanilla JS and styles: 

Sandbox: https://codesandbox.io/s/affectionate-fermi-d62vvb?file=/index.html
Preview: https://d62vvb.csb.app/

Leaving this open so I can remember to file a bug tomorrow.

**Comment by thecrypticace** [comment]
Okay after a bit of sleuthing this morning I discovered there's an already existing bug that was described to have a very similar behavior. In that instance it was noted as no longer reproducible so I've commented on the bug report with the reproduction details. Hopefully they'll be able to use that to take action on the underlying problem. Closing as there's not much we can do here unfortunately :(

Thank you for bringing the bug to our attention so we could provide them with more details.

Here's the bug report: https://bugs.chromium.org/p/chromium/issues/detail?id=1274326#c23
